pkg/scheduler/ugm/manager.go

```diff
@@ -401,6 +390,27 @@ func (m *Manager) clearEarlierSetLimits(newUserLimits map[string]map[string]*Lim
 	m.clearEarlierSetUserLimits(newUserLimits)
 }
 
+// clearEarlierSetUserWildCardLimits Traverse new wild card user config and decide whether earlier usage needs to be cleared or not
+// by comparing with the existing config. If config set earlier not now, then traverse all users, check whether wild card limit has been applied/used or not.
+// Reset earlier settings for the users only when wild card user limit has been applied.
+func (m *Manager) clearEarlierSetUserWildCardLimits(newUserWildCardLimits map[string]*LimitConfig) {
+	for queuePath := range m.userWildCardLimitsConfig {
+		hierarchy := strings.Split(queuePath, configs.DOT)
+		// Is queue path exists?
+		if _, ok := newUserWildCardLimits[queuePath]; !ok {
+			for u, ut := range m.userTrackers {
+				// Has wild card user limit applied for this user?
+				if m.HasWildCardApplied(u, hierarchy) {
```

Comment:
Here we find out if it has a wildcard applied and then in line 407 we do the work again to remove the settings. Why are we not just doing the reset directly?

Reply:
Taken care